stat/distuv/poisson.go

```diff
@@ -0,0 +1,97 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distuv
+
+import (
+	"math"
+	"math/rand"
+
+	"gonum.org/v1/gonum/mathext"
+)
+
+// Poisson implements the Poisson distribution, a discrete probability distribution
+// that expresses the probability of a given number of events occurring in a fixed
+// interval of time and/or space.
+//
+// The poisson distribution has density function:
+//
+//  λ^k / k! e^(-λ)
+//
+// For more information, see https://en.wikipedia.org/wiki/Poisson_distribution
+type Poisson struct {
+	// Lambda is the average number of events in an interval.
+	// Lambda must be greater than 0.
+	Lambda float64
+
+	Source *rand.Rand
+}
+
+// CDF computes the value of the cumulative distribution function at x.
+func (p Poisson) CDF(x float64) float64 {
+	if x < 0 {
+		return 0
+	}
+	kfac := 1
+	ix := int(math.Floor(x))
+	for i := 1; i <= ix; i++ {
+		kfac *= i
+	}
+	return mathext.GammaInc(math.Floor(x+1), p.Lambda) / float64(kfac)
+}
+
+// Entropy returns the entropy of the distribution.
```

Comment:
Remove the functions that are not implemented. When this is through, could you add a PR with the unimplemented functions?

Reply:
done.